src/DynamoCore/Core/UndoRedoRecorder.cs

```diff
@@ -448,21 +448,27 @@ public void Dispose()
             }
         }
 
-        internal class NodeModificationUndoHelper : IDisposable
+        internal class ModelModificationUndoHelper : IDisposable
         {
-            private readonly NodeModel node;
+            private readonly List<ModelBase> models;
             private readonly UndoRedoRecorder recorder;
             private readonly Dictionary<Guid, XmlElement> existingConnectors;
             private readonly Dictionary<Guid, ConnectorModel> remainingConnectors;
 
-            public NodeModificationUndoHelper(UndoRedoRecorder recorder, NodeModel node)
+            public ModelModificationUndoHelper(UndoRedoRecorder recorder, ModelBase model)
+                : this(recorder, new [] { model })
```

Comment:
Good question there, @aparajit-pratap! The `NodeModificationUndoHelper` was initially designed to deal with anything that has connectors (which turns out to be just `NodeModel`). The idea of that class was to help record node modification which can eventually result in dropping connectors. Now that we want to apply model update to a selection set, which can potentially include `NoteModel`, this class then gets extended to work with the base class `ModelBase`. Does that make sense?